ntp.go

```diff
@@ -42,28 +42,8 @@ func (t ntpTime) UTC() time.Time {
 	return time.Date(1900, 1, 1, 0, 0, 0, 0, time.UTC).Add(time.Duration(t.nsec()))
 }
 
-func (t ntpTime) nsec() uint64 {
-	return uint64(t.Seconds) * 1e9 + (uint64(t.Fraction) * 1e9 >> 32)
-}
-
-func (a ntpTime) subtract(b ntpTime) ntpTime {
-	var result ntpTime
-	result.Fraction = a.Fraction - b.Fraction
-	result.Seconds = a.Seconds - b.Seconds
-	if result.Fraction > a.Fraction {
-		result.Seconds -= 1
-	}
-	return result
-}
-
-func (r ntpTime) add(a ntpTime) ntpTime {
-	var result ntpTime
-	result.Fraction = r.Fraction + a.Fraction
-	result.Seconds = r.Seconds + a.Seconds
-	if result.Fraction < a.Fraction {
-		result.Seconds += 1
-	}
-	return result
+func (t ntpTime) nsec() int64 {
+	return int64(t.Seconds) * 1e9 + int64(float64(t.Fraction) / frac * 1e9)
```

Comment:
I'm not sure why you changed the calculation to do a conversion to and from floating point here.  What was wrong with the original method?